### ISSUE TITLE
Vertical bar | within $$..$$ breaks equation

### ISSUE THREAD

**Issue by dinh-toan** [alshedivat/al-folio#2189]
### Have you checked that your issue isn't already filed?

- [X] I read through [FAQ](https://github.com/alshedivat/al-folio/blob/master/FAQ.md) and searched through the [past issues](https://github.com/alshedivat/al-folio/issues), none of which addressed my issue.
- [X] Yes, I have checked that this issue isn't already filed.

### Bug description

Markdown rendering problem with vertical bar '|' inside math delimiters '$$'

### Description
Every time I need to write an inline math expression with a vertical bar |, for example  `abc $$m|n$$ def `
It seems that al-folio does not recognize math delimiters, but instead thinks I am trying to write a markdown table. Moreover, the problem worsens when there are multiple math expressions within a single paragraph.

### How to reproduce the issue
1. Open any markdown post.
2. Insert in any line an inline math expression with |, for example:
`This theme supports rendering beautiful $$\mid |$$  ... inside a paragraph, it will produce an inline expression $$m^2$$....`
3.  Expected behaviour:

> This theme supports rendering beautiful $\mid |$  ... inside a paragraph, it will produce an inline expression $m^2$....

4. Resulted bevaviour:
```
<table>
<tbody>
 <tr>
<td>This theme supports rendering beautiful $$\mid</td>
<td>\(... inside a paragraph, it will produce an inline expression\)m^2$$…</td>
</tr>
</tbody>
</table>
```

### Illustrated images
_Code_ 
<img width="742" alt="Screenshot 2024-02-13 at 11 53 32 AM" src="https://github.com/alshedivat/al-folio/assets/159132711/9c2a09f7-ce3d-40da-9433-d041fd6e04d2">
_Site_
<img width="802" alt="Screenshot 2024-02-13 at 11 53 46 AM" src="https://github.com/alshedivat/al-folio/assets/159132711/ccf00cf6-958e-4c55-b9ca-cf9356c0e204">

### What operating system are you using?

Mac

### Where are you seeing the problem on?

Deployed site

**Comment by george-gca** [comment]
Using it as `$$\mid$$` it renders as `|`.

**Comment by dinh-toan** [comment]
Thanks and it does not render double bars. Another example for my case is writing the absolute value. More precisely, `$$|p|$$` stills breaks equation.

**Comment by george-gca** [comment]
Note that this symbol can be generated in [numerous ways](https://latex-programming.fandom.com/wiki/Vertical-bar_(LaTeX_symbol)), so just try a different one.

But what you are trying to achieve has more adequate ways in LaTeX. I recommend using an [online tool](https://latex.codecogs.com/eqneditor/editor.php) to help you build your equations.

**Comment by dinh-toan** [comment]
Is it possible to address the parsing problem in Al Folio? I think ensuring robust math rendering is a crucial aspect of Al Folio's functionality.

**Comment by george-gca** [comment]
`$$\|a\|$$` renders `||a||`.

Also, are you sure only `|a|` is a valid LaTeX equation? If so, I am afraid we may be with our hands tied here, since we rely on [MathJax](https://www.mathjax.org/) to convert LaTeX equations.

**Comment by dinh-toan** [comment]
Yes, `|a|` is indeed valid LaTex code. I believe we are not with our hands tied here as this code renders correctly in MathJax—I've confirmed this using the MathJax Live demo available at https://www.mathjax.org/#demo.

In my opinion, this issue might stem from Al Folio automatically converting inline delimiters from `$$..$$` to `\( ..\)` before employing MathJax for math expression rendering. Perhaps, I'm wrong.

**Comment by george-gca** [comment]
Hummm interesting. Did a [quick search](https://talk.jekyllrb.com/t/jekyll-and-mathjax/5514) on this, and maybe it can be customized inside MathJax, since it seems to be a [limitation of kramdown](https://kramdown.gettalong.org/syntax.html#math-blocks), which is the markdown flavor of jekyll. But I don't think this is worth the trouble, since kramdown docs itself states:

> Note that LaTeX code that uses the pipe symbol | in inline math statements may lead to a line being recognized as a table line. This problem can be avoided by using the \vert command instead of |!

**Comment by dinh-toan** [comment]
Thank you Geogre.
